src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -111,6 +115,9 @@ object SnowflakeTelemetry {
       metric.put(TelemetryOOBFields.EXCEPTION_MESSAGE, "NA")
       metric.put(TelemetryOOBFields.EXCEPTION_STACKTRACE, "NA")
     }
+    if (config.isDefined) {
+      metric.set(TelemetryOOBFields.SYSTEM_CONFIG, config.get)
```

Comment:
Use config.foreach

Reply:
With “config.foreach”, the spark option will be in current level (not a sub level). When we query the message, we may see spark option names are delimited by “.”, SC option names are delimited “_”. Does this concern you?

Reply:
I had misunderstanding for config.foreach. I will update it.

Reply:
I tried to use “config.foreach”, but it doesn’t work in this case. I think it is nice to have. Let’s just use “IF”.
config.foreach(println(_)) // This works
config.foreach(x => metric.set(TelemetryQueryStatusFields.SYSTEM_CONFIG, x)) // This raises exception.

The exception is “net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.node.ObjectNode cannot be cast to scala.runtime.Nothing$”